### PR TITLE
Add assert into the insertBlockHeader

### PR DESCRIPTION
**Pull request by Mr-Leshiy**

### CODE REVIEW

File: include/veriblock/blockchain/alt_block_tree.hpp [hunk 1]
```diff
@@ -316,6 +316,9 @@ struct AltBlockTree : public BaseBlockTree<AltBlock> {
   void removeAllPayloads(index_t& index);
 };
 
+template <>
+void assertInsertBlockHeader(const AltBlock& block);
```
Comment (Warchant) [inline include/veriblock/blockchain/alt_block_tree.hpp:320]:
```suggestion
void assertBlockSanity(const AltBlock& block);
```

File: src/blockchain/alt_block_tree.cpp [hunk 2]
```diff
@@ -663,6 +663,13 @@ std::vector<const AltBlockTree::index_t*> AltBlockTree::getConnectedTipsAfter(
   return candidates;
 }
 
+template <>
+void assertInsertBlockHeader(const AltBlock& block) {
+  VBK_ASSERT_MSG(
+      block.getHash() != block.getPreviousBlock(),
+      "previous block hash should be different with the current block hash");
```
Comment (Warchant) [inline src/blockchain/alt_block_tree.cpp:670]:
```suggestion
      "Previous block hash should not be equal to the current block hash=%s. A collision in altchain hash?, block.getHash()");
```

File: include/veriblock/blockchain/blockchain_util.hpp [hunk 3]
```diff
@@ -124,6 +123,9 @@ bool recoverEndorsements(ProtectedBlockTree& ed_,
 template <typename Block>
 void assertBlockCanBeRemoved(const Block& block);
 
+template <typename Block>
+void assertInsertBlockHeader(const Block& block);
```
Comment (Warchant) [inline include/veriblock/blockchain/blockchain_util.hpp:127]: you can add default implementation and specify only VBK variant

File: include/veriblock/blockchain/alt_block_tree.hpp (hunk 1 above)
Comment (Mr-Leshiy) [reply include/veriblock/blockchain/alt_block_tree.hpp]: fixed

File: src/blockchain/alt_block_tree.cpp (hunk 2 above)
Comment (Mr-Leshiy) [reply src/blockchain/alt_block_tree.cpp]: fixed

File: include/veriblock/blockchain/blockchain_util.hpp (hunk 3 above)
Comment (Mr-Leshiy) [reply include/veriblock/blockchain/blockchain_util.hpp]: fixed

File: include/veriblock/blockchain/blockchain_util.hpp [hunk 4]
```diff
@@ -124,6 +123,14 @@ bool recoverEndorsements(ProtectedBlockTree& ed_,
 template <typename Block>
 void assertBlockCanBeRemoved(const Block& block);
 
+template <typename Block>
+void assertBlockSanity(const Block& block) {
+  VBK_ASSERT_MSG(block.getHash() != block.getPreviousBlock(),
+                 "Previous block hash should NOT be equal to the current block "
+                 "hash: %s. A collision in altchain hash?",
+                 HexStr(block.getHash()).c_str());
```
Comment (Warchant) [inline include/veriblock/blockchain/blockchain_util.hpp:131]:
```suggestion
                 HexStr(block.getHash()));
```

File: src/blockchain/pop/vbk_block_tree.cpp [hunk 5]
```diff
@@ -450,6 +450,14 @@ void assertBlockCanBeRemoved(const BlockIndex<VbkBlock>& index) {
                  index.endorsedBy.size());
 }
 
+template <>
+void assertBlockSanity(const VbkBlock& block) {
+  VBK_ASSERT_MSG(block.getShortHash() != block.getPreviousBlock(),
+                 "Previous block hash should NOT be equal to the current block "
+                 "hash: %s. A collision in altchain hash?",
+                 HexStr(block.getShortHash()).c_str());
```
Comment (Warchant) [inline src/blockchain/pop/vbk_block_tree.cpp:458]:
```suggestion
                 HexStr(block.getShortHash()));
```

**Review by Warchant — APPROVED** [review]

File: src/blockchain/pop/vbk_block_tree.cpp (hunk 5 above)
Comment (Mr-Leshiy) [reply src/blockchain/pop/vbk_block_tree.cpp:458]: fixed

File: include/veriblock/blockchain/blockchain_util.hpp (hunk 4 above)
Comment (Mr-Leshiy) [reply include/veriblock/blockchain/blockchain_util.hpp]: fixed

**Pull request merged**